### PR TITLE
Issue#21-22: login logout

### PR DESCRIPTION
**Pull request by Anmavel**

### CODE REVIEW

File: frontend/src/component/Navigation.tsx [hunk 1]
```diff
@@ -1,16 +1,50 @@
...
-            <div className="navbar">
-                <NavLink to="/" className="active"><i className="fa fa-fw fa-home"></i> Home</NavLink>
-                <NavLink to="/sign-up"><i className="fa fa-fw fa-user"></i>SignUp</NavLink>
-            </div>
+            {user ?
+                <>
+                    <link rel="stylesheet"
+                          href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"/>
+                    <div className="navbar">
+                        <NavLink to="/" className="active"><i className="fa fa-fw fa-home"></i> Home</NavLink>
```
Comment (valfink) [inline frontend/src/component/Navigation.tsx:33]: Also the Home link :)

File: frontend/src/component/PetGallery.tsx [hunk 2]
```diff
@@ -8,13 +8,13 @@ import useAuth from "../hooks/useAuth";
...
 type PetGalleryProps = {
     pets: Pet[]
-
     navigateTo: string | undefined
 }
 
 export default function PetGallery(props: PetGalleryProps) {
     const user = useAuth(true)
-    const pets = props.pets.map((pet: Pet) => {
+    const filteredPets = props.pets.filter(pet => pet.userId === user?.id);
```
Comment (valfink) [inline frontend/src/component/PetGallery.tsx:16]: I suggest filtering the pets in the backend and only returning the user's own pets to the frontend

**Review by valfink — APPROVED** [review]
Good work! :)

File: backend/src/main/java/com/example/backend/controller/MongoUserController.java [hunk 3]
```diff
@@ -17,9 +17,17 @@ public class MongoUserController {
...
+        return getMe(principal);
+    }
     @GetMapping("/me")
     public MongoUserResponse getMe(Principal principal) {
         return mongoUserDetailsService.getMe(principal);
     }
+    @PostMapping("/logout")
+    public void logout() {
+        // this method exists only to define the endpoint!
+    }
```
[the remark this answers is not in the document]
Comment (Anmavel) [reply backend/src/main/java/com/example/backend/controller/MongoUserController.java:31]: Good to know 👍

File: backend/src/test/java/com/example/backend/controller/MongoUserControllerTest.java [hunk 4]
```diff
@@ -60,4 +61,59 @@ void postUser_whenNoCSRFToken_then403() throws Exception {
...
+                .andExpect(content().json("""
+                        {
+                        "username": "user",
+                        "role": "BASIC"
+                        }
+                                                """));
+        mockMvc.perform(MockMvcRequestBuilders.post("/api/users/logout").with(csrf()))
+                .andExpect(status().isOk());
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/users/me"))
+                .andExpect(status().isUnauthorized());
```
[the remark this answers is not in the document]
Comment (Anmavel) [reply backend/src/test/java/com/example/backend/controller/MongoUserControllerTest.java:115]: thanks

File: backend/src/test/java/com/example/backend/exception/GlobalExceptionHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+import java.util.Map;
+
+class GlobalExceptionHandlerTest {
+
+    @Test
+    @ExceptionHandler(UnauthorizedException.class)
```
[the remark this answers is not in the document]
Comment (Anmavel) [reply backend/src/test/java/com/example/backend/exception/GlobalExceptionHandlerTest.java:14]: oh... I don´t think so, I think you are right and I don´t need it here. Thanks

File: backend/src/test/java/com/example/backend/exception/UnauthorizedExceptionTest.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+class UnauthorizedExceptionTest {
+    @Test
+    void testUnauthorizedExceptionWithDefaultMessage() {
+        UnauthorizedException exception = Assertions.assertThrows(UnauthorizedException.class,
+                () -> {
+                    throw new UnauthorizedException();
+                });
```
[the remark this answers is not in the document]
Comment (Anmavel) [reply backend/src/test/java/com/example/backend/exception/UnauthorizedExceptionTest.java:12]: thanks. ok, then i will only create a new instance of the exception and check the message 👍

File: backend/src/test/java/com/example/backend/exception/UnauthorizedExceptionTest.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        Assertions.assertEquals("You are not authorized!", exception.getMessage());
+    }
+
+    @Test
+    void testUnauthorizedExceptionWithCustomMessage() {
+        String message = "You are not authorized to access this Pet!";
+        UnauthorizedException exception = Assertions.assertThrows(UnauthorizedException.class,
+                () -> {
+                    throw new UnauthorizedException(message);
+                });
```
[the remark this answers is not in the document]
Comment (Anmavel) [reply backend/src/test/java/com/example/backend/exception/UnauthorizedExceptionTest.java:23]: 👍

File: frontend/src/component/Navigation.tsx [hunk 8]
```diff
@@ -1,16 +1,50 @@
...
     return (
         <nav>
-            <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"/>
-            <div className="navbar">
-                <NavLink to="/" className="active"><i className="fa fa-fw fa-home"></i> Home</NavLink>
-                <NavLink to="/sign-up"><i className="fa fa-fw fa-user"></i>SignUp</NavLink>
-            </div>
+            {user ?
+                <>
+                    <link rel="stylesheet"
```
[the remark this answers is not in the document]
Comment (Anmavel) [reply frontend/src/component/Navigation.tsx:30]: 👍

File: frontend/src/component/Navigation.tsx (hunk 1 above)
Comment (Anmavel) [reply frontend/src/component/Navigation.tsx:33]: 👍

**Pull request merged**